VENTokenField/VENBackspaceTextField.m

```diff
@@ -24,15 +24,34 @@
 
 @implementation VENBackspaceTextField
 
-- (BOOL)keyboardInputShouldDelete:(UITextField *)textField
-{
-    if (self.text.length == 0) {
-        if ([self.delegate respondsToSelector:@selector(textFieldDidEnterBackspace:)]) {
-            [self.delegate textFieldDidEnterBackspace:self];
+- (BOOL)keyboardInputShouldDelete:(UITextField *)textField {
+    BOOL shouldDelete = YES;
+
+    if ([UITextField instancesRespondToSelector:_cmd]) {
+        BOOL (*keyboardInputShouldDelete)(id, SEL, UITextField *) = (BOOL (*)(id, SEL, UITextField *))[UITextField instanceMethodForSelector:_cmd];
+
+        if (keyboardInputShouldDelete) {
+            shouldDelete = keyboardInputShouldDelete(self, _cmd, textField);
         }
     }
 
-    return YES;
+    if (![textField.text length] && [[[UIDevice currentDevice] systemVersion] intValue] >= 8) {
+        [self deleteBackward];
+    }
+
+    return shouldDelete;
+}
+
+- (void)deleteBackward {
```

Comment:
This however I can find in the API and has been there since iOS 3.2. I'm guessing it won't work without the private API method override above, but if it could be done without it it would be fore the best :P